app/models/batch.rb

```diff
@@ -0,0 +1,21 @@
+class Batch < ApplicationRecord
+
+  belongs_to :branch
+  belongs_to :coaching
+  has_many :registrations, dependent: :restrict_with_error
+  has_many :players, through: :registrations
+
+  validates :coach_name, :players_limit, :start_at_in_HH, :end_at_in_HH, presence:  true
+  validates :players_limit, :players_count, numericality: {
+    only_integer: true,
+    greater_than_or_equal_to: 0
```

Comment:
what does limit 0 mean?

Reply:
there should be an option to give no limit

Reply:
Presence validator removed for players limit, so now Players limit can be left blank to have infinite players or must be greater than 10.